### ISSUE TITLE
Assembler: Jetpack maps on pages don't render randomly

### ISSUE THREAD

**Issue by miksansegundo** [Automattic/wp-calypso#86785]
Jetpack maps on pages have a rendering issue because after refreshing the browser a few times, they don't render.

Initially, they render, but it seems a second render breaks them.

https://github.com/Automattic/wp-calypso/assets/1881481/4b9144cb-08dc-4e2c-aec6-41240344b7f3

When the page is re-rendered by unmarking /marking the checkbox, the map renders well.

https://github.com/Automattic/wp-calypso/assets/1881481/fd385455-9f80-4f0b-aa4b-c01555104a63

There are some warnings in the console but no errors:

![Image](https://github.com/Automattic/wp-calypso/assets/1881481/216a10da-7327-487f-836f-0f6db4cbc9b6)

**Comment by miksansegundo** [comment]
@Automattic/lego, any ideas of what's happening here?

Wondering if we can avoid multiple renders of page patterns, for example, only re-render when there are changes in colors/fonts.

**Comment by fushar** [comment]
I spent the whole day investigating this issue without luck. 😅 😭 

Some observations:

1. This issue only happens if we refresh the browser in the "Add more pages" screen. I cannot reproduce this in the "normal" usage of the Assembler flow.
2. This pattern's html changes every time we fetch the rendered pattern. See screenshot:

<img width="1439" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1525580/af2e04a7-927d-4417-9ddd-ca67f53045ee">

It seems to come from Akismet. However, I have tried stripping this Akismet-related hidden input, but the issue still persist...

3. Multiple rerenders is inevitable upon browser reload, because we need to call `useRenderedPatterns()`, which fires multiple requests to wpcom (because we fetch the patterns by category).
4. I tried to "shove" `useMemo()`, `React.memo()` everywhere I can to reduce the number of rerenders. Still no luck.

Given (1), I think we can deprioritize this issue. At least, I'll stop investigating for now. 🥲
